### DISCUSSION TITLE
Feature Change Suggestion

### DISCUSSION THREAD

**Discussion by Uplinkpro** [retronas/discussions#32] [Ideas]
Finally have this setup working in docker on unraid. Its a very nice bit of kit and everything so far works great however I would like to make a suggestion on how the "Rom Import" scripts works . As it is as I Understand, It scans all roms in the romimports folder against premade lists, then if the exists moves/symlinks them into the proper locations for the roms to work on mister and in the general roms folder. After one list finishes it outputs a file noting what you are missing and the missing files checksum.

My requested changes are small but would make a huge impact on how this function works. When starting the script have it check for a missing file.txt for each system Before it scans for said system, if the file exists skip that scan. And secondly after moving/linking files, remove them from the romimport folder, if they stay in said folder they are rescanned for every system causing the scanning to take very long and most times ending in a crash. It would also let us see if the premade lists are possibly missing some files they should not be. 

These changes would be huge in making this function not only usable but less resource and time intensive.

**Comment by sairuk** [comment]
we are leveraging functionality from https://github.com/frederic-mahe/Hardware-Target-Game-Database the features you are after would be better served built in to the main tool and then we can leverage them in a menu.

If we were to do something in our bash wrapper it'd be a menu item to clear the folder the user would manually trigger. We "don't delete things" preferring to leave that decision to user but we could provide something from the interface to make it easier to clean the folder. We don't have access to the processed files as they go through, hence the functionality on a per file basis would need to be added to the main project.

Is the assumption with skipping when a missing.txt is present that a scan has already occurred when the file is present? that won't capture if the user adds additional files to the folder, they would need to remove the missing.txt to allow for rescan.

happy to see a pull request for what you're thinking if anyone is up to it, otherwise when we get time some good ideas there.

**Reply by Uplinkpro** [reply]
Ah ok, that makes since, ill make a post about that on his github about getting that function added. As for the other bit, Yes its like you suggested, it would be a simple on off switch, if the person wanted the scan to re run they would remove the txt file and it would re run. I understand this can be done via the CLI with a targeted system,  however, like this you can just hit the button in tools and run multiple  as long as they did not have the txt files present.

**Comment by qcretro** [comment]
I just started checking out Retronas. I have most of the SMDBs filled out already in my collection. Importing them via the tool has been a long process. I have an idea that would help speed up the process without going into how the htgdb tool works, I think. There could be a submenu for the tool in Retronas that lets the user check/uncheck which system SMDB to process for. I first tried dumping all my roms in the romimport folder and letting the tool run but that is super long and inefficient because it is scanning 10000s of files for every console SMDB it is checking. Lots of hash to compute, again and again... not so long for older 8bit consoles, but CD ISOs are much longer to process. I've found it faster to add one system at the time to the romimport folder and run the tool each time. Inefficient again, but already much faster. Selecting which files to process would greatly help speeding up the import process IMO.
